h2/src/main/org/h2/mvstore/FreeSpaceBitSet.java

```diff
@@ -111,12 +111,13 @@ public long allocate(int length) {
      * Allocate a number of blocks and mark them as used.
      *
      * @param length the number of bytes to allocate
-     * @param reservedLow start position of the reserved are
-     * @param reservedHigh end position of the reserved area
+     * @param reservedLow start block index of the reserved area (inclusive)
+     * @param reservedHigh end block index of the reserved area (exclusive),
+     *                     special value -1 means beginning of the infinite free area
      * @return the start position in bytes
      */
     long allocate(int length, long reservedLow, long reservedHigh) {
-        return getPos(allocate(getBlockCount(length), getBlock(reservedLow), getBlock(reservedHigh), true));
+        return getPos(allocate(getBlockCount(length), (int)reservedLow, (int)reservedHigh, true));
```

Comment:
If using type int for block index, here type cast is unnecessary.

Reply:
It's not the end of the world, just little annoying.